src/errors/extractErrors.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+'use strict';
+
+import fs from 'fs';
+import path from 'path';
+import * as babylon from 'babylon';
+import traverse from 'babel-traverse';
+import { invertObject } from './invertObject';
+import { evalToString } from './evalToString';
+import { paths } from '../constants';
+import { safeVariableName } from '../utils';
+import pascalCase from 'pascal-case';
+
+const babylonOptions = {
+  sourceType: 'module',
+  // As a parser, babylon has its own options and we can't directly
+  // import/require a babel preset. It should be kept **the same** as
+  // the `babel-plugin-syntax-*` ones specified in
+  // https://github.com/facebook/fbjs/blob/master/packages/babel-preset-fbjs/configure.js
+  plugins: [
+    'classProperties',
+    'flow',
+    'jsx',
+    'trailingFunctionCommas',
+    'objectRestSpread',
+  ],
+};
+
+export function extractErrors(opts: any) {
+  if (!opts || !('errorMapFilePath' in opts)) {
+    throw new Error(
+      'Missing options. Ensure you pass an object with `errorMapFilePath`.'
+    );
+  }
+
+  if (!opts.name || !('name' in opts)) {
+    throw new Error('Missing options. Ensure you pass --name flag to tsdx');
+  }
+
+  const errorMapFilePath = opts.errorMapFilePath;
+  let existingErrorMap: any;
+  try {
+    // Using `fs.readFileSync` instead of `require` here, because `require()`
+    // calls are cached, and the cache map is not properly invalidated after
+    // file changes.
+    existingErrorMap = JSON.parse(
+      fs.readFileSync(
+        path.join(__dirname, path.basename(errorMapFilePath)),
```

Comment:
this is a bug, we should just read from `errorMapFilePath`, exactly how we write.